lib/yabeda/sidekiq.rb

```diff
@@ -59,6 +60,16 @@ module Sidekiq
           sidekiq_queue_latency.set({ queue: queue.name }, queue.latency)
         end
 
+        now = Time.zone.now
+        ::Sidekiq::Workers.new.each do |process, thread, msg|
+          payload = msg['payload']
+
+          sidekiq_worker_runtime.set(
+            {queue: payload['queue'], worker: payload['class']},
+            now - Time.at(msg['run_at'])
+          )
+        end
```

Comment:
What if process executes several jobs of one worker class? Which value will be recorded?

The problem here is that last thread value will be used, and last value is always the most recent one: https://github.com/mperham/sidekiq/blob/f9f3959c344c5bfcfad6793104f743a4296e2e90/lib/sidekiq/api.rb#L939

So you want not last, but maximum of the values (so first of them).

You can either traverse it in reverse order or group them.

I propose grouping:
```suggestion
        ::Sidekiq::Workers.new.each_with_object(Hash.new { |h, k| h[k] = 0 }) do |result, (_process, _thread, msg)|
          payload = msg['payload']
          tags = { queue: payload['queue'], worker: payload['class'] }
          duration = now - Time.at(msg['run_at'])
          result[tags] = duration if result[tags] < duration
        end.each do |tags, duration|
          sidekiq_worker_runtime.set(tags, duration)
        end
```